src/main/java/frc/team2412/robot/subsystem/DrivebaseSubsystem.java

```diff
@@ -210,10 +214,24 @@ public double getAngularVelocity() {
         }
     }
 
+    public static final double P = 0.01, THRESHOLD = 5;
+
+    private double defaultX, defaultY;
+
     public void drive(Vector2 translationalVelocity, double rotationalVelocity, boolean isFieldOriented) {
         isFieldOrientedEntry.setBoolean(isFieldOriented);
         synchronized (stateLock) {
-            driveSignal = new HolonomicDriveSignal(translationalVelocity, rotationalVelocity, true);
+            if (isFieldOriented) {
+                synchronized (sensorLock) {
+                    double xAdj = gyroscope.getAxis(Pigeon.Axis.ROLL) - defaultX,
+                            yAdj = gyroscope.getAxis(Pigeon.Axis.PITCH) - defaultY;
+                    driveSignal = new HolonomicDriveSignal(translationalVelocity.rotateBy(gyroscope.getAngle())
+                            .add(Math.abs(xAdj) > THRESHOLD ? xAdj * P : 0, Math.abs(yAdj) > THRESHOLD ? yAdj * P : 0),
+                            rotationalVelocity, false);
```

Comment:
Please refactor this to be more readable for the benefit of reviewers and future readers.